hardware-handler/nightmare_hardware_handler/src/hardware-handler.py

```diff
@@ -13,48 +15,83 @@
 from std_msgs.msg import Header
 
 # robot servo serial ports
-tty_list = ["/dev/ttyUSB0", "/dev/ttyUSB1", "/dev/ttyUSB2", "/dev/ttyUSB3"]
+TTY_LIST = ["/dev/ttyUSB0", "/dev/ttyUSB1", "/dev/ttyUSB2", "/dev/ttyUSB3"]
+NUMBER_OF_LEGS = 6
+NUMBER_OF_SERVOS = 19
+NUMBER_OF_SERVOS_PER_LEG = 3
 
 
 # simple range mapping function
 def fmap(x: float, in_min, in_max, out_min, out_max):
     return (x - in_min) * (out_max - out_min) / (in_max - in_min) + out_min
 
 
-# servo class to hold servo parameters
+@dataclass
 class Servo:
-    def __init__(self):
-        self.tty = ""
-        self.id = 0
-        self.angle = 0
-        self.pos = 0
+    """servo class to hold servo parameters"""
+    id: int
+    tty: str
```

Comment:
No initialization?

Reply:
Nope, you can't have a default id and a tty, those are known at instantiation